manual/parts/threads.tex

```diff
@@ -559,19 +559,36 @@ \subsection{Debug Exception: Single-stepping}
 \subsection{Timeout Fault (MCS only)}
 \label{sec:timeout-fault}
 
-Timeout faults are raised when a thread consumes all of its budget and has a
-timeout fault handler that is not a null capability.  They allow a timeout
-exception handler to take some action to restore the thread, and deliver a
-message containing the scheduling context data word, as well as the amount of
-time consumed since the last timeout fault occured on this scheduling context,
-or since \apifunc{seL4\_SchedContext\_YieldTo}{schedcontext_yieldto} or
+Timeout faults are raised when a thread loses access to time with which
```

Comment:
Good start, however I recommend some changes for readability. 

* You've dropped the important and still relevant part of the first sentence: " has a
￼timeout fault handler that is not a null capability"
* Use a bulleted list for all of the reasons you may get a timeout fault or break into subsections . If this is too big and there are nuances,introduce the topic softly then go into detail on each one. this allows people to get the gist without reading, and also allows them to get to the section they want to know about very quickly. 

```
Timeout faults are raised when a thread attempts to execute but cannot as it has no access to time, *and* if the thread has a valid timeout fault handler. Timeout faults fall into three categories: budget exhaustion, missing scheduling contexts, or invalid scheduling contexts.
```
Note the above maps to the 3 fault types.

Now follow with three subheadings and details + examples of each type of fault.

Reply:
(Note for myself):

* [ ] Use list for fault types
* [ ] Add section describing *exhausted* SC reason w/ examples
* [ ] Add section describing *no* SC reason w/ examples
* [ ] Add section describing *unconfigured* SC reason w/ examples